### PR TITLE
[11.0][IMP] Total of all rows in web_widget_x2many_2d_matrix

### PR DESCRIPTION
**Pull request by tarteo**
This PR will add a cell to the right corner with the total of all the columns / rows.

image

### CODE REVIEW

**Review by astirpe — APPROVED** [review]
We are using this improvement in production (with module hr_timesheet_sheet) and it works fine.

**Review by pedrobaeza — APPROVED** [review]

File: web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js [hunk 1]
```diff
@@ -304,12 +304,26 @@ odoo.define('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer', function (requ
...
+                var $tr = $('<tr>').append('<td/>').append($cells);
+                var $total_cell = this._renderTotalCell();
+                if ($total_cell) {
+                    $tr.append($total_cell);
+                }
+                return $('<tfoot>').append($tr);
             }
         },
 
+        _renderTotalCell: function() {
```
Comment (simahawk) [inline web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js]: docstring pls :)

**Review by simahawk — CHANGES_REQUESTED** [review]
nitpicking, otherwise LGTM:
1. docstring missing
2. can you update the screenshot in README?

**Review by simahawk — APPROVED** [review]
thanks!

**Pull request merged**